src/Library/Library.fs

```diff
@@ -1,127 +1,90 @@
-﻿namespace Library
+﻿namespace MyListSorts
 
 
-module factorial =
-    let rec factorial n =
-        if n < 0 then
-            failwith "The factorial of negative numbers is not calculated by this program\n"
-        elif n = 0 || n = 1 then
-            1
-        else
-            n * factorial (n - 1)
-
-
-module sorts =
-
-    let swap (arr: 't[]) i j =
-        let elem = arr[i]
-        arr[i] <- arr[j]
-        arr[j] <- elem
-
-
-    let bubble_sort (arr: 't[]) : 't[] =
-
-        let mutable array_len = arr.Length
-        let mutable swaped = true
-
-        while swaped do
-            swaped <- false
-
-            for i in 0 .. array_len - 2 do
-                if compare arr[i] arr[i + 1] > 0 then
-                    swap arr i (i + 1)
-                    swaped <- true
-
-        array_len <- array_len - 1
-        arr
-
+type MyList<'t> =
+    | Empty
+    | Cons of 't * MyList<'t>
 
-    let partition (arr: 't[]) (low: int) (high: int) =
-        let pivot = arr[high]
-        let mutable i = low - 1
 
-        for j in low .. (high - 1) do
-            if compare arr[j] pivot <= 0 then
-                i <- i + 1
-                swap arr i j
+module MyList =
 
-        swap arr (i + 1) high
-        i + 1
+    let rec toMyList (list: 't List) : (MyList<'t>) =
+        match list with
+        | [] -> Empty
+        | head :: tail -> Cons(head, toMyList tail)
 
-    let rec inplace_quick (arr: 't[]) (low: int) (high: int) =
-        if low < high then
-            let pivotIndex = partition arr low high
-            inplace_quick arr low (pivotIndex - 1)
-            inplace_quick arr (pivotIndex + 1) high
+    let rec toList (list: MyList<'t>) : ('t List) =
+        match list with
+        | Empty -> []
+        | Cons(head, tail) -> head :: toList tail
 
-    let quick_sort (arr: 't[]) : 't[] =
-        inplace_quick arr 0 (Array.length arr - 1)
-        arr
 
 
-    let merge (arr: 't[]) (low: byref<int>) (high: byref<int>) (mid: byref<int>) =
-        let mutable start2 = mid + 1
-
-        if compare arr[mid] arr[start2] > 0 then
-            while low <= mid && start2 <= high do
-                if compare arr[low] arr[start2] <= 0 then
-                    low <- low + 1
+    let rec bubbleSort (list: MyList<'t>) : MyList<'t> =
```

Comment:
Whether this function should be recursive?

Reply:
bubbleSort does not have to be recursive. But it is called in the last line of code. I can call just bubble instead of bubbleSort